fix timed_command on items with autoupdate=false

jimtng opened the pull request:
The problem: 
My items are mqtt based with autoupdate=false. When I send a command to the item, it sends it out via mqtt, but doesn't immediately updates its state. Its state only gets updated by an incoming MQTT message from the device.

Meanwhile, the timed command had already set its change detection rule. When the state update (for the original command) came back from MQTT, it fired the change detection rule which cancels the timed command.

This PR prevents that by restricting the change detection rule to trigger only when the previousState == command, so it would not be triggered by this initial change of state caused by the timed command itself.

~~There is still another edge case that this PR doesn't address:~~
~~- When you issued a timed command from one rule, e.g. Item1.on for: 5.minutes~~
~~- Another rule issued Item1.on afterwards~~

~~You'd probably expect that the timed command is cancelled, and Item1 would stay on. I don't think it's the case right now. An additional received_command trigger is needed.~~

The edge case above has been addressed in this PR too

ccutrer commented on lib/openhab/dsl/items/timed_command.rb:
```diff
@@ -197,7 +201,10 @@ def initialize(timed_command_details)
-              config: { "itemName" => timed_command_details.item.name }
+              config: {
+                "itemName" => timed_command_details.item.name,
+                "previousState" => timed_command_details.command.to_s
```
you should run this through `timed_command_details.item.format_command(command).to_s`.

jimtng replied:
Good catch!

ccutrer commented on lib/openhab/dsl/items/timed_command.rb:
```diff
@@ -30,6 +30,8 @@ module TimedCommand
         #
         # @attr [Item] item
         #   @!visibility private
+        # @attr [Object] command
```
it's probably best to label this as accepting a Core::Types::Command, not an Object. technically, yes, it can be an object, but that's rarely helpful in documentation. then we should probably add docs to Command (and State, for GenericItem#update) that most places a Command (or Type) is accepted, types that are convertible to the proper type for the applicable item are also accepted.

jimtng replied:
+1

ccutrer commented on lib/openhab/dsl/items/timed_command.rb:
```diff
@@ -42,6 +44,7 @@ module TimedCommand
         #   @!visibility private
         #
         TimedCommandDetails = Struct.new(:item,
+                                         :command,
```
there should be no need to persist the command on the timed command details, just send it as a param to create_timed_command

jimtng replied:
I was actually thinking, why not officially expose/document these things to the block. It may be useful to make a decision in choosing the reverted state based on the original command and item. WDYT?

ccutrer replied:
there's nothing in here that _should_ be available to the block, and isn't already available when the block is created

jimtng replied:
What if the block is a proc defined elsewhere?

jimtng replied:
I've just removed it from the Struct and passed it as an argument.

jimtng commented:
Addressed change requests (except the inclusion of :command in the TimedCommandDetails - pending feedback)
Added the received command trigger + related tests.
